### DISCUSSION TITLE
Middleware redirect

### DISCUSSION THREAD

**Discussion by Ali-Hussein-dev** [vercel/next.js#38587] [Help] [answered]
I have an issue when redirecting users to the login page if they are not authenticated. 
When pathname doesn't start with "/auth", I redirect it to "/auth", but it shows only a blank page because all paths that include the static files of "/auth" will be false and so will not be sent with the response.

```ts
import { NextRequest, NextResponse } from 'next/server';

export const middleware = (req: NextRequest) => {
  const { cookies,  nextUrl } = req;

  const isAuthenticated = cookies.get('isLoggedIn');

  const pathname = nextUrl.pathname;

  if (!isAuthenticated) {
    const loginUrl = new URL('/auth', req.url);
    if (!pathname.startsWith('/auth')) { 
    return NextResponse.redirect(loginUrl);
    } else {
     return NextResponse.next(); 
    }
    
  }

  if (isAuthenticated) {
  // ...
  }
};
```
Next version `12.2.2`

**Comment by icyJoseph** [comment, marked as answer]
Perhaps you want to add this:

```ts
const PUBLIC_FILE = /\.(.*)$/;

export function middleware(req: NextRequest) {
  const { pathname } = req.nextUrl;

  if (
    pathname.startsWith("/_next") || // exclude Next.js internals
    pathname.startsWith("/api") || //  exclude all API routes
    pathname.startsWith("/static") || // exclude static files
    PUBLIC_FILE.test(pathname) || // exclude all files in the public folder
  )
    return NextResponse.next();

   // then comes your implementation
```

I am not sure I understand the problem, but I think this is it.

**Reply by Ali-Hussein-dev** [reply]
yes, it is right. Thank for your help. 
But shouldn't be handled by Next?

**Reply by icyJoseph** [reply]
Not with redirects, only rewrites I think, or what do you mean `handled by Next`? 

In the latest release they made it so that the internals are also seen by the middleware, https://nextjs.org/docs/messages/middleware-upgrade-guide#executing-middleware-on-internal-nextjs-requests.

**Reply by Ali-Hussein-dev** [reply]
I mean like `rewrite`, just providing the pathname, without thinking about internal nextjs requests. 
Where could executing middleware on the internal nextjs requests be beneficial? What are the use cases?
